flytekit/types/directory/types.py

```diff
@@ -144,7 +145,9 @@ def extension(cls) -> str:
     def __class_getitem__(cls, item: typing.Union[typing.Type, str]) -> typing.Type[FlyteDirectory]:
         if item is None:
             return cls
-        item_string = str(item)
+
+        item_string = FileExt.check_and_convert_to_str(item)
```

Comment:
@eapolinario this is what I'm refering to, since the FlyteDirectory takes in `Annotated[str, FileExt()]`